### PR TITLE
feat(libs/backendless-console-api): add NSOrder, NSFFMTItem to tables prefixes to ignore 

### PR DESCRIPTION
**Pull request by mcntai**
Changes:

- parametrize `columnsToIgnore` and `tablesToIgnore`
- add line break for Roles Permissions diff printing
- fix bug which prevented from deleting local apps that are not in schema
- enrich column with columnId where needed as this is the only way to update system columns

it deletes and prints the table in app if it's not in schema

image

### CODE REVIEW

File: libs/backendless-console-api.js [hunk 1]
```diff
@@ -496,6 +493,8 @@ class Backendless {
       (table.relations || []).forEach(cleanRelation);
     }
 
+    app.tables = app.tables.filter(table => !tablesToIgnore.find(prefix => table.name.startsWith(prefix)))
```
Comment (vengrov) [inline libs/backendless-console-api.js]: Do this once in `getAppDataTables` please and forget about this option in any other places

File: libs/comparator/app-permissions.js [hunk 2]
```diff
@@ -51,8 +51,7 @@ const printDifferences = (apps, appRolesMap) => {
     })
 
     if (result) {
-        console.log('\nRoles Permissions:')
-        console.log(table.toString())
+        console.log('\nRoles Permissions:\n' + table.toString())
```
Comment (vengrov) [inline libs/comparator/app-permissions.js:54]: Does not it break a table layout?

File: libs/comparator/tables.js [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 
-const SYSTEM_COLUMNS = ['created', 'updated', 'ownerId', 'objectId', 'blUserLocale']
-
-const buildColumnsMap = table => {
+const buildColumnsMap = (table, columnsToIgnore) => {
     const result = {}
 
     table.columns.forEach(column => {
-        if (!SYSTEM_COLUMNS.includes(column.name)) {
+        if (!columnsToIgnore.includes(column.name)) {
```
Comment (vengrov) [inline libs/comparator/tables.js]: no need to do any filtering on this level as we already did it right after fetching the table metadata

File: libs/index.js [hunk 4]
```diff
@@ -42,26 +42,26 @@ module.exports = options => {
     .then(() => checkList[API_PERMS] && backendless.getAppServicesRolePermissions())
     .then(() => checkList[API_KEYS] && backendless.getAppCustomApiKeys())
     .then(() => apps = backendless.getApps())
-    .then(() => dumpPath && BackendlessConsole.dump(apps[0], dumpPath, verboseOutput))
+    .then(() => dumpPath && BackendlessConsole.dump(apps[0], dumpPath, verboseOutput, tablesToIgnore))
```
Comment (vengrov) [inline libs/index.js]: getAppDataTables should be the only place where tablesToIgnore is used

File: libs/sync/tables.js [hunk 5]
```diff
@@ -87,6 +92,8 @@ const syncColumn = async (api, app, tableName, columnName, sourceColumn, targetC
                     await bulkUpdate(api, app, tableName, where, { [columnName]: sourceColumn.defaultValue })
                 }
 
+                sourceColumn.columnId = targetColumn?.columnId
```
Comment (vengrov) [inline libs/sync/tables.js:95]: This change looks suspicious :suspect:
What is it for?

File: libs/sync/tables.js [hunk 6]
```diff
@@ -161,11 +168,8 @@ const syncColumns = (api, apps, opts) => {
...
-        .then(() => syncTables(api, apps, opts))
-        // update table data
-        .then(() => api.getAppDataTables())
-        .then(() => syncColumns(api, apps, opts))
-        .then(() => cleanup(api, apps))
\ No newline at end of file
+   syncTables(api, apps, opts)
+      .then(() => api.getAppDataTables(opts.columnsToIgnore))
+      .then(() => syncColumns(api, apps, opts))
+      .then(() => cleanup(api, apps))
```
Comment (vengrov) [inline libs/sync/tables.js]: If you decided to refactor it, why not replace all these constructions with async/await?

```suggestion
  await syncTables(api, apps, opts)
  await api.getAppDataTables(opts.columnsToIgnore)
  await syncColumns(api, apps, opts)
  await cleanup(api, apps)
```

**Review by vengrov — CHANGES_REQUESTED** [review]

File: libs/comparator/app-permissions.js (hunk 2 above)
Comment (mcntai) [reply libs/comparator/app-permissions.js:54]: no, it fixes it.

before changes

image


after changes 

image

File: libs/sync/tables.js (hunk 5 above)
Comment (mcntai) [reply libs/sync/tables.js:95]: @vengrov system columns can be updated only if there is `columnId` in payload, so I added it.

should I use some other approach for handling system columns update ?

**Pull request merged**